bpf/lib/nodeport.h

```diff
@@ -1548,6 +1606,16 @@ static __always_inline int dsr_reply_icmp4(struct __ctx_buff *ctx,
 	if (ctx_store_bytes(ctx, off + sizeof(ip), &icmp,
 			    sizeof(icmp), 0) < 0)
 		goto drop_err;
+	if (ctx_store_bytes(ctx, off + sizeof(ip) + sizeof(icmp),
+				&inner_ip_hdr, sizeof(inner_ip_hdr), 0) < 0)
+		goto drop_err;
+	if (inner_ip_hdr.protocol == IPPROTO_UDP ||
+		inner_ip_hdr.protocol == IPPROTO_TCP) {
+		if (ctx_store_bytes(ctx, off + sizeof(ip) + sizeof(icmp)
+				+ sizeof(inner_ip_hdr) + l4_dport_offset,
```

Comment:
Would would happen to the csum of the the ICMP packet, I think if you change the dport it will have to change too, no?

Reply:
Right. I have now corrected this issue. I will add a new comment showing ipv4/icmpv4, ipv6/icmpv6 packet responses which has correct checksum.